src/app/components/MediaLoader/configs/portraitClipMedia.ts

```diff
@@ -23,77 +22,102 @@ export default ({
   adsEnabled = false,
   showAdsBasedOnLocation = false,
 }: ConfigBuilderProps): ConfigBuilderReturnProps => {
-  const portraitClipMediaBlock: PortraitClipMediaBlock = filterForBlockType(
+  const firstBlock = filterForBlockType(
     blocks,
     'portraitClipMedia',
-  );
-
-  const { images, video, type } = portraitClipMediaBlock?.model || {};
-
-  const { source, urlTemplate: locator } = images?.[1] ?? {};
-
-  const originCode = source?.replace('Image', '');
-
-  const versionID = video?.version?.id;
-
-  const clipISO8601Duration = video?.version?.duration;
-
-  const rawDuration = moment.duration(clipISO8601Duration).asSeconds();
-
-  const title = video?.title;
-
-  const videoId = video?.id;
-
-  const captionBlock = getCaptionBlock(blocks, 'live');
-
-  const caption =
-    captionBlock?.model?.blocks?.[0]?.model?.blocks?.[0]?.model?.text;
-
-  const kind = video?.version?.kind || 'programme';
-
-  const guidanceMessage = video?.version?.guidance;
+  ) as PortraitClipMediaBlock;
+
+  if (!firstBlock) {
```

Comment:
Originally when trying to refactor the `filterForBlockType` we split the file into two functions to preserve the existing logic. I think this `firstBlock` is still needed to get the right metadata for the playlist but we're open to ideas here